docs/java/features/extensions/extension-library/curconv/currency-conversion.md

```diff
@@ -13,69 +13,225 @@ keywords:
+
 ```java
-//example code
-```
+TenantAccessor.executeWithTenant(providerTenant, CallableWhichPerformsCurrencyConversion);
 
-#### Use Integration Objects
-Use integration objects that contain data models and UIs, and services that operate on the data models that you can use to send information to the Currency Conversion library.
+TenantAccessor.executeWithTenant(() -> MyProviderTenant, () -> {
+  // Call the Currency Conversion library here.
+});
+```
+## Code Example
+The code example that follows, shows the end-to-end process flow in using the Currency Conversion library.
 
 ```java
-//example code
-```
+// Initialize the Currency Conversion Library.
+CurrencyConverter currConverter = new CurrencyConverter();
 
-#### Multi-tenancy with Currency Conversion
-Based on the dependency added to pom.xml, the library identifies the environment you are operating on and identifies the tenant context. The library automatically derives the tenant context by using the Tenant abstraction provided by the SAP Cloud SDK. To override the default conversion configuration, manually add an exchange rate provider and an exchange rate provider code to the API request, as follows. If you do this, the default configuration settings are not taken into consideration.
+// Initialize the Data Adapter implementation you would like to use.
+DataAdapter dataAdapter = new DataAdapterImpl();
 
-```java
-TenantAccessor.executeWithTenant(providerTenant, CallableWhichPerformsCurrencyConversion);
+// Prepare the parameter for your conversion request.
+ConversionParametersForNonFixedRate parameter =
+    ConversionParametersForNonFixedRate.builder()
+      .fromAmount(new CurrencyAmount("500.123"))
+      .fromCurrency(Currency.getInstance("EUR"))
+      .toCurrency(Currency.getInstance("USD"))
+      .exchangeRateType(new RateType("MID"))
+      .build();
 
-TenantAccessor.executeWithTenant(() -> “MyProviderTenant”, () -> {
-  // Call currency conversion library here.
-});
-```
-## Demo App using Currency Conversion
+// Call the conversion library for a single conversion.
+
+try {
+      SingleNonFixedRateConversionResult singleConversionresult =
+        currConverter.convertCurrencyWithNonFixedRate(param, dataAdapter);
+      convertedAmount =
+        singleConversionresult.getConvertedAmount().getDecimalValue();
+} catch(ConversionException ex){
+      slfLogger.error("Conversion exception with message as " + ex.getMessage());
```

Comment:
Leave the catch block empty with three dots "..." and a comment like "Exception handling here"

Reply:
The current exception handling is not optimal, like throwing another exception based on the caught exception...